rules-reviewed/eap7/eap6/hsearch.windup.xml

```diff
@@ -0,0 +1,1153 @@
+                    <link
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/#migrate_lucene_renamed_and_repackaged_classes"
+                        title="Lucene - Renamed and Repackaged Classes" />
+                </hint>
+            </perform>
+        </rule>
+<!--     o.a.l.analysis.ASCIIFoldingFilter -> o.a.l.analysis.miscellaneous.ASCIIFoldingFilter -->
+<!--     o.a.l.analysis.ISOLatin1AccentFilter -> o.a.l.analysis.miscellaneous.ISOLatin1AccentFilter -->
+<!--     o.a.l.analysis.KeywordMarkerFilter -> o.a.l.analysis.miscellaneous.KeywordMarkerFilter -->
+<!--     o.a.l.analysis.LengthFilter -> o.a.l.analysis.miscellaneous.LengthFilter -->
+<!--     o.a.l.analysis.PerFieldAnalyzerWrapper -> o.a.l.analysis.miscellaneous.PerFieldAnalyzerWrapper -->
+        <rule id="hsearch-00229">
+            <when>
+                <javaclass references="org.apache.lucene.analysis.{type}" />
+            </when>
+            <perform>
+                <hint title="Renamed package for {type} in Lucene 4.x" effort="1" severity="mandatory">
+                    <message>Lucene's core and contrib analyzers, along with Solr's analyzers, were consolidated into lucene/analysis module.
+                     During the refactoring package name have changed for `org.apache.lucene.analysis.{type}` to `org.apache.lucene.analysis.miscellaneous.{type}`.</message>
+                    <link href="http://lucene.apache.org/core/4_10_2/MIGRATE.html" title="Apache Lucene Migration Guide" />
+                    <link href="http://hibernate.org/search/documentation/migrate/5.0/#within-apache-lucene" title="Renamed classes within Apache Lucene" />
+                    <link
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/#migrate_lucene_renamed_and_repackaged_classes"
+                        title="Lucene - Renamed and Repackaged Classes" />
+                </hint>
+            </perform>
+            <where param="type">
+                <matches pattern="(ASCIIFoldingFilter|ISOLatin1AccentFilter|KeywordMarkerFilter|LengthFilter|PerFieldAnalyzerWrapper)" />
+            </where>
+        </rule>
+<!--     o.a.l.analysis.TeeSinkTokenFilter -> o.a.l.analysis.sinks.TeeSinkTokenFilter -->
+        <rule id="hsearch-00230">
+            <when>
+                <javaclass references="org.apache.lucene.analysis.TeeSinkTokenFilter" />
+            </when>
+            <perform>
+                <hint title="Renamed package for TeeSinkTokenFilter in Lucene 4.x" effort="1" severity="mandatory">
+                    <message>Lucene's core and contrib analyzers, along with Solr's analyzers, were consolidated into lucene/analysis module.
+                
```

Comment:
`TokenizerFactory` is good, `TokenizerFilterFactory` is bad. It's `TokenFilterFactory` -- it's also written in the comment above the rule.